js/whitebit.js

```diff
@@ -503,42 +572,69 @@ module.exports = class whitebit extends Exchange {
         //         "isBuyerMaker":false
         //     }
         //
+        // orderTrades (v4Private)
+        //       {
+        //          "time": 1593342324.613711,      // Timestamp of executed order
+        //          "fee": "0.00000419198",         // fee that you pay
+        //          "price": "0.00000701",          // price
+        //          "amount": "598",                // amount in stock
+        //          "id": 149156519,                // trade id
+        //          "dealOrderId": 3134995325,      // completed order Id
+        //          "clientOrderId": "customId11",  // custom order id; "clientOrderId": "" - if not specified.
+        //          "role": 2,                      // Role - 1 - maker, 2 - taker
+        //          "deal": "0.00419198"            // amount in money
+        //       }
         let timestamp = this.safeValue (trade, 'time');
         if (typeof timestamp === 'string') {
             timestamp = this.parse8601 (timestamp);
         } else {
             timestamp = parseInt (timestamp * 1000);
         }
-        const priceString = this.safeString (trade, 'price');
-        const amountString = this.safeString2 (trade, 'amount', 'volume');
-        const cost = this.parseNumber (Precise.stringMul (priceString, amountString));
-        const price = this.parseNumber (priceString);
-        const amount = this.parseNumber (amountString);
+        const orderId = this.safeString (trade, 'dealOrderId');
+        const cost = this.safeString (trade, 'deal');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString2 (trade, 'amount', 'volume');
         const id = this.safeString2 (trade, 'id', 'tradeId');
         let side = this.safeString (trade, 'type');
         if (side === undefined) {
             const isBuyerMaker = this.safeValue (trade, 'isBuyerMaker');
-            side = isBuyerMaker ? 'buy' : 'sell';
+            if (side !== undefined) {
+                side = isBuyerMaker ? 'buy' : 'sell';
+            }
         }
-        let symbol = undefined;
-        if (market !== undefined) {
-            symbol = market['symbol'];
+        const symbol = this.safeSymbol (undefined, market);
+        let quoteId = undefined;
+        if (symbol !== undefined) {
+            quoteId = symbol.split ('/')[1];
```

Comment:
We should not do any symbol parsing in the parser, all parsing has to be in fetchMarkets.
In the parsers we use the already-preloaded info about the markets, matching it by market id.
Also, we never return non-unified exchange-specific market ids and currency ids from the parsers to the user.
We should always transform those to the unified symbols and currency codes with `safeSymbol`, `safeMarket` or `safeCurrencyCode` (check mentions of those across the other exchanges in the lib).

If we need the market structure to get the quote currency, we should do the following:

```JavaScript
const market = this.safeMarket (undefined, market);
const quote = market['quote'];
// ...
fee = {
    'cost': feeCost,
    'currency': quote,
};
```

Reply:
my bad going to change it now